ethereum/contracts/EtherVault.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Holds ETH on behalf of ETHApp
+contract EtherVault is Ownable {
+    event Deposit(address account, uint256 amount);
+    event Withdraw(address account, address recipient, uint256 amount);
+
+    // Accepts ETH from the caller.
+    receive() external payable {
```

Comment:
There is nothing that prevents someone from sending eth directly to the vault, rather than going through the channel. This is fine from a security perspective, but I think bad due to the following:
 - UX issue -> people will likely mistakenly send eth to the vault expecting something to happen when they need to go via channel
 - Accounting issue -> there will be a mismatch between eth in the vault and wrapped eth on polkadot, so annoying/messy from an accounting perspective

We should probably make this onlyOwner too?

Reply:
> There is nothing that prevents someone from sending eth directly to the vault, rather than going through the channel. This is fine from a security perspective, but I think bad due to the following:
> 
> * UX issue -> people will likely mistakenly send eth to the vault expecting something to happen when they need to go via channel

This is already an issue in all our app contracts. Someone sending Ether to our EtherApp will lose it forever. Same with ERC20App, and DotApp.

> * Accounting issue -> there will be a mismatch between eth in the vault and wrapped eth on polkadot, so annoying/messy from an accounting perspective
> 

Not really possible to prevent this. There is no way to reliable way to prevent a contract from receiving Ether: https://consensys.github.io/smart-contract-best-practices/development-recommendations/general/force-feeding/

> We should probably make this onlyOwner too?

I'd rather not, because it would imply we're trying to prevent unauthorized deposits, when in fact that's not technically possible.

Reply:
On accounting issue - yeah fair, not possible to prevent in theory but in practice you can prevent everything except self destruct, so i do think that's still worthwhile.

On UX issue - yeah then we should ideally be fixing this issue across all our contracts, not just this one. It is very, very likely that if we don't fix this, we'll end up with random people messing up and giving us operational tickets to get their funds back due to it. I've seen this happen on almost every other project that accepts tokens/eth directly.

Reply:
> I'd rather not, because it would imply we're trying to prevent unauthorized deposits, when in fact that's not technically possible.

We can always just add a comment that acknowledges this and explains that it's just to prevent transfers for UX reasons, not for security purposes and that self-destructs are still fine.

Reply:
Sounds good :+1:

@alistair-singh yeah, so put a `revert` in the `receive` function, and then add an explicit payable `deposit` function only callable by the app. Would be good to emit a `Deposit` event with the app address and the actual owner of the funds.